### PR TITLE
[casclib] Update to 2024-06-05

### PR DESCRIPTION
**Pull request by Shauren**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Instead of patching, it could just use `-DCMAKE_REQUIRE_FIND_PACKAGE_ZLIB=ON`.

**Comment by Shauren** [comment]
Interesting, I did not know about that cmake feature

File: ports/casclib/portfile.cmake [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+endif()
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        -DCMAKE_REQUIRE_FIND_PACKAGE_ZLIB=ON
         -DCASC_BUILD_SHARED_LIB=${CASC_BUILD_SHARED_LIB}
         -DCASC_BUILD_STATIC_LIB=${CASC_BUILD_STATIC_LIB}
-        -DCASC_UNICODE=ON
+        -DCASC_UNICODE=${CASC_UNICODE}
```
Comment (dg0yt) [inline ports/casclib/portfile.cmake:26]: Isn't this changing the behavior for existing users?

File: ports/casclib/portfile.cmake [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" CASC_BUILD_SHARED_LIB)
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" CASC_BUILD_STATIC_LIB)
 
+set(CASC_UNICODE OFF)
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    message(STATUS "This version of CascLib is built in ASCII mode. To switch to UNICODE version, create an overlay port of this with CASC_UNICODE set to ON.")
+    message(STATUS "This recipe is at ${CMAKE_CURRENT_LIST_DIR}")
+    message(STATUS "See the overlay ports documentation at https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md")
+endif()
```
Comment (dg0yt) [inline ports/casclib/portfile.cmake:18]: Note that the effect can be achieved by triplet customization, adding `-DCASC_UNICODE=<whatever>` to `VCPKG_CMAKE_CONFIGURE_OPTIONS` in a per-port section.
https://learn.microsoft.com/en-us/vcpkg/users/triplets#per-port-customization

Comment (Shauren) [reply ports/casclib/portfile.cmake:18]: That might be true, I copied this messages from maintainer guide https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#recommended-solutions

File: ports/casclib/portfile.cmake (hunk 1 above)
Comment (Shauren) [reply ports/casclib/portfile.cmake:26]: It does but I have no better solution

CASC_UNICODE only does something on windows (it uses TCHAR in public api, manually defining it to char on other platforms) which leads to inconsistent behavior for consumers of this port if they write applications that can be compiled on more platforms

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**